contrib/pkg/createcluster/create.go

```diff
@@ -266,13 +266,13 @@ func (o *Options) GenerateObjects() ([]runtime.Object, error) {
 	if err != nil {
 		return nil, err
 	}
-	pullSecret, err := o.generatePullSecret()
+	sshSecret, err := o.generateSSHSecret()
 	if err != nil {
 		return nil, err
 	}
-	sshSecret, err := o.generateSSHSecret()
+	pullSecret, err := o.generatePullSecret()
 	if err != nil {
-		return nil, err
+		return append([]runtime.Object{awsCreds, sshSecret}, result...), nil
```

Comment:
Because pull secret is optional, it should not throw any error if pull secret is not provided.

Reply:
Optional does not imply that errors are not errors. For example, if a pull secret file is supplied but there is an error reading the file, that is an error that needs to bubble up. That should not cause the pull secret file to be ignored.

Reply:
That needs to be handled in `hiveutil` binary. Currently we use `hiveutil` to generate the yaml files and `hiveutil` looks for `.pull-secret`. So any error ends up in the yaml file. I need to look in to that. Till now I was assuming that that something we can handle later.

Reply:
I don't understand this. I would expect that if there is a problem reading the pull secret file that the user supplied, then the `create-cluster` command would fail. I would not expect that the `create-cluster` command would create a yaml file that is missing the pull secret object as though the user did not supply a pull secret file.

Reply:
Something should be done for create-cluster at some point, right now it assumes there must always be a pull secret and tries to load the default location, and indeed if not found it ignores this and still tries to create an empty pull secret.

I'm open to doing this as a followup provided this PR can go in without it in a good state, if it becomes important we should do it in this one.

Reply:
And I agree, the error here should probably not be swallowed, better to fail the command.

Reply:
Agree, I will revert the changes and will handle this in a followup PR